### DISCUSSION TITLE
Solara interactive components don't visualize in PyCharm Jupyter notebooks

### DISCUSSION THREAD

**Discussion by EwoutH** [widgetti/solara#354] [Q&A]
_This is not necessarily a Solara bug, I think the problem lies with JetBrain's their implementation of Jupyter Notebooks, but I think it could be handy to have it documented and discussed here._

_I'm also curious if other JetBrains PyCharm (or DataSpell) users can reproduce this bug._

> When running Solara interactive components in PyCharm, it doesn't provide the inline visualisation that's expected.
> 
> Demo notebook not working in PyCharm, to reproduce: [demo.ipynb](https://youtrack.jetbrains.com/api/files/74-2030680?sign=MTY5ODc5NjgwMDAwMHwxMS05MjEwODB8NzQtMjAzMDY4MHxlSV9xS2JXWFNsY1g4N0l6bldVQUkzVVJIaFVQNlRraGowczY4cFBzeHhrDQo&updated=1698146596250&forceDownload=true)
> 
> ![Screenshot_176](https://github.com/widgetti/solara/assets/15776622/3b2ad46c-216c-4f01-b579-bdf3d6948ab2)
> 
> Here it's on Google Colab, and it visualizes properly: https://colab.research.google.com/drive/1fjd3KQevsRmHl9ky95zIjaTrhKSOzGxS?usp=sharing
> 
> ![Screenshot_177](https://github.com/widgetti/solara/assets/15776622/7e4e1942-6e18-4529-8eb4-3dd46851609f)
> 
> Here's also a pure Python example that does work in PyCharm (download both in same folder, run in command line with solara run app.py): [app.py](https://youtrack.jetbrains.com/api/files/74-2030694?sign=MTY5ODc5NjgwMDAwMHwxMS05MjEwODB8NzQtMjAzMDY5NHxTVGNOOEE4TmZaWVppSGFzWWR1clFFeUR1NDg0NEJ1bUNkVmZYcm42VWtRDQo&updated=1698147025578&forceDownload=true) + [model.py](https://youtrack.jetbrains.com/api/files/74-2030693?sign=MTY5ODc5NjgwMDAwMHwxMS05MjEwODB8NzQtMjAzMDY5M3xvMnh1ZjBnbUNqQ245ZU81TktlT25xS1FwWDJMMDc4RmROVDRXX1pJazdzDQo&updated=1698147025573&forceDownload=true). The issue is the visualisation in the Jupyter notebooks.
> 
> PY-232.9559.58, JRE 17.0.8+7-b1000.8x64 JetBrains s.r.o., OS Windows 11(amd64) v10.0 , screens 6720.0x4200.0

See the full bug here on JetBrain YouTrack: [Solara interactive components don't visualize in PyCharm Jupyter notebooks](https://youtrack.jetbrains.com/issue/PY-63760/Solara-interactive-components-dont-visualize-in-PyCharm-Jupyter-notebooks)

**Comment by EwoutH** [comment]
I can confirm that this issue is fixed starting from PyCharm 2023.3.3 RC!

**Reply by maartenbreddels** [reply]
Nice, thanks for letting us know. I followed the conversation at youtrack, but getting a user-confirmation is best :)
